### PR TITLE
Implement functionality for the searchbars.

### PR DESCRIPTION
**Pull request by KochTobi**
Implement functionality for the searchbars.

This PR implements the functionality for the search bars. Entered text is now searched for in project code or title as well as in sample code or title.

### CODE REVIEW

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectFilterImpl.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  public ProjectFilter allowEmptyProjects(boolean emptyProjectAllowed) {
+    this.emptyProjectAllowed = emptyProjectAllowed;
+    return this;
+  }
+
+  @Override
+  public Predicate<? extends ProjectSummary> asPredicate() {
+
+    return containsText().and(it -> emptyProjectAllowed || it.getTotalSampleCount() > 0);
+  }
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectFilterImpl.java:43]: What is an empty project?
If I recall correctly total sample count > 0 is currently set directly as a filter on the grid. I don't think we need to add it here. I think removing those, in general, is a precondition for the content in the gird and the other is custom filtering for what the user types in. I would prefer to keep those separate

File: sample-tracking-status-overview-app/src/main/resources/life/qbic/portal/sampletracking/components/sampleoverview/SampleDesign.html [hunk 2]
```diff
@@ -40,4 +40,4 @@
    </vaadin-grid>
   </vaadin-vertical-layout>
  </body>
-</html>
\ No newline at end of file
+</html>
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-app/src/main/resources/life/qbic/portal/sampletracking/components/sampleoverview/SampleDesign.html:43]: 👌🏼

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectFilterImpl.java (hunk 1 above)
Comment (KochTobi) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectFilterImpl.java:43]: The previous solution always overwrote my filters. This way it is all handled by the filter.

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleFilterImpl.java [hunk 3]
```diff
@@ -22,25 +24,42 @@
...
+  public SampleFilter containingText(String substring) {
+    this.substring = substring;
+    return this;
+  }
 
+  @Override
+  public Predicate<? extends Sample> asPredicate() {
+    final Predicate<Sample> containsText = it -> {
+      if (substring.isEmpty())
+        return true;
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleFilterImpl.java:47]: Either put the if statement in brackets (I recall you taught me its how you should do it) or maybe remove it? You did not have this check in the project, I would prefer to have the same style for both of them :)

**Review by jenniferboedker — COMMENTED** [review]
Nice work!
NIT: maybe add a CL entry for the searchbars since they are new features

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleFilterImpl.java (hunk 3 above)
Comment (jenniferboedker) [inline sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleFilterImpl.java:47]:
```suggestion
    final Predicate<Sample> containsText = it -> {substring.isEmpty()
```

**Review by jenniferboedker — DISMISSED** [review]

**Review by jenniferboedker — DISMISSED** [review]

**Review by jenniferboedker — APPROVED** [review]

File: sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleFilterImpl.java (hunk 3 above)
Comment (KochTobi) [reply sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleFilterImpl.java]: good catch :+1:

**Pull request merged**